src/main/java/net/snowflake/ingest/streaming/internal/ParquetRowBuffer.java

```diff
@@ -205,9 +203,17 @@ private float addRow(
       RowBufferStats forkedStats = statsMap.get(columnName).forkEmpty();
       forkedStatsMap.put(columnName, forkedStats);
       ColumnMetadata column = parquetColumn.columnMetadata;
-      ParquetValueParser.ParquetBufferValue valueWithSize =
-          ParquetValueParser.parseColumnValueToParquet(
-              value, column, parquetColumn.type, forkedStats, defaultTimezone, insertRowsCurrIndex);
+      ParquetBufferValue valueWithSize =
+          (column.getSourceIcebergDataType() == null
```

Comment:
In iceberg mode we expect every column to have sourceIcebergDataType populated.
Lets:
1. Add a property to ClientBufferParameters called isIcebergMode, and populate it via the clientInternal.getInternalParameterProvider().getIsIcebergMode().
2. use that boolean to decide which ValueParser to use here instead of column.getSourceIcebergDataType()

3. Wherever appropriate - assert that column.getSourceIcebergDataType is ALWAYS not-null when we're in iceberg mode. This will likely be in channel_open response handling

Reply:
Added `isIcebergMode` in ClientBufferParameters and added check in SnowflakeStreamingIngestClientInternal.openChannel.